File: src/code/python/sorting_algorithms/bubble_sort.py
def bubble_sort(nums: list) -> None:
    n = len(nums)

    for i in range(n):
        swapped = False

        for j in range(0, n - i - 1):
            if nums[j] > nums[j + 1]:
                nums[j], nums[j + 1] = nums[j + 1], nums[j]
                swapped = True

        if not swapped:
            break

File: src/code/ruby/sorting_algorithms/bubble_sort.rb
def bubble_sort(arr)
  n = arr.size

  n.times do |i|
    swapped = false

    (0...(n - i - 1)).each do |j|
      if arr[j] > arr[j + 1]
        arr[j], arr[j + 1] = arr[j + 1], arr[j]
        swapped = true
      end
    end

    break unless swapped
  end
end
